cloud/amazon/iam.py

```diff
@@ -548,6 +557,22 @@ def main():
     new_path = module.params.get('new_path')
     key_count = module.params.get('key_count')
     key_state = module.params.get('access_key_state')
+    trust_policy = module.params.get('trust_policy')
+
+    apdoc=None
+
+    if trust_policy != None:
+        try:
+            json_data = open(trust_policy, 'r')
+        except IOError:
+            try:
+                apdoc = json.dumps(trust_policy)
+            except Exception as e:
+                module.fail_json(msg=str(e) + '\n' + module.params.get('trust_policy'))
+        else:
+            apdoc = json.dumps(json.load(json_data))
```

Comment:
This fails to catch badly formatted json causing an uncapatured exception to be thrown

```shell
Traceback (most recent call last):
  File "/Users/jdavila/.ansible/tmp/ansible-tmp-1457729969.51-237985966879139/iam", line 3057, in <module>
    main()
  File "/Users/jdavila/.ansible/tmp/ansible-tmp-1457729969.51-237985966879139/iam", line 573, in main
    apdoc = json.dumps(json.load(json_data))
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 291, in load
    **kw)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

Def want to capture this and give a sane error message to the user.